lib/csv/table.rb

```diff
@@ -432,9 +462,41 @@ def headers
     #
     # ---
     #
-    # The expression <tt>table[header]</tt>, where +header+ is a \String,
-    # returns column values (\Array of \Strings) if the column exists
-    # and if the access mode is <tt>:col</tt> or <tt>:col_or_row</tt>:
+    # Fetch Columns by \Range::
+    # - Form: <tt>table[range]</tt>, +range+ a \Range object.
+    # - Access mode: <tt>:col</tt>.
+    # - Return value: column data from the table, beginning at column <tt>range.start</tt>,
+    #   if those columns exist.
+    #
+    # Returns column values from the table, if the column exists;
+    # the values are arranged by row:
+    #   source = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+    #   table = CSV.parse(source, headers: true)
+    #   table.by_col!
+    #   table[0..1] # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+    #
+    # Special case: if <tt>range.start == headers.size</tt>,
+    # returns an \Array (size: <tt>table.size</tt>) of empty \Arrays:
+    #   table[table.headers.size..50] # => [[], [], []]
+    #
+    # If <tt>range.end</tt> is negative, calculates the ending index from the end:
+    #   table[0..-1] # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+    #
+    # If <tt>range.start</tt> is negative, calculates the starting index from the end:
+    #   table[-2..2] # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+    #
+    # If <tt>range.start</tt> is larger than <tt>table.size</tt>,
+    # returns an \Array of +nil+ values:
+    #   table[4..4] table[4..4]
```

Comment:
Could you update the result value?

Reply:
Thanks for catching this.  Fixed in new push.